src/java.base/share/classes/jdk/internal/vm/vector/VectorSupport.java

```diff
@@ -185,82 +186,131 @@ public VectorMask(Object payload) {
+                    case 16: obj = new VectorPayloadMF128B(); break;
+                    case 32: obj = new VectorPayloadMF256B(); break;
+                    case 64: obj = new VectorPayloadMF512B(); break;
+                    default: assert false : "Unhandled vector size";
+                }
+            } else if (elemType == short.class) {
+                switch(length) {
+                    case  4: obj = new VectorPayloadMF64S(); break;
+                    case  8: obj = new VectorPayloadMF128S(); break;
+                    case 16: obj = new VectorPayloadMF256S(); break;
+                    case 32: obj = new VectorPayloadMF512S(); break;
+                    default: assert false : "Unhandled vector size";
+                }
+            } else if (elemType == int.class) {
+                switch(length) {
+                    case  2: obj = new VectorPayloadMF64I(); break;
+                    case  4: obj = new VectorPayloadMF128I(); break;
+                    case  8: obj = new VectorPayloadMF256I(); break;
+                    case 16: obj = new VectorPayloadMF512I(); break;
+                    default: assert false : "Unhandled vector size";
+                }
+            } else if (elemType == long.class) {
+                switch(length) {
+                    case  1: obj = new VectorPayloadMF64L(); break;
+                    case  2: obj = new VectorPayloadMF128L(); break;
+                    case  4: obj = new VectorPayloadMF256L(); break;
+                    case  8: obj = new VectorPayloadMF512L(); break;
+                    default: assert false : "Unhandled vector size";
+                }
+            } else if (elemType == float.class) {
+                switch(length) {
+                    case  2: obj = new VectorPayloadMF64F(); break;
+                    case  4: obj = new VectorPayloadMF128F(); break;
+                    case  8: obj = new VectorPayloadMF256F(); break;
+                    case 16: obj = new VectorPayloadMF512F(); break;
+                    default: assert false : "Unhandled vector size";
+                }
+            } else if (elemType == double.class) {
+                switch(length) {
+                    case  1: obj = new VectorPayloadMF64D(); break;
+                    case  2: obj = new VectorPayloadMF128D(); break;
+                    case  4: obj = new VectorPayloadMF256D(); break;
+                    case  8: obj = new VectorPayloadMF512D(); break;
+                    default: assert false : "Unhandled vector size";
+                }
             }
             return obj;
         }
 
-        public static VectorPayloadMF createVectPayloadInstance(int elemSize, int length) {
-            return newInstanceFactory(elemSize, length);
+        @ForceInline
+        public static VectorPayloadMF createVectPayloadInstance(Class<?> elemType, int length) {
+            return newInstanceFactory(elemType, length);
```

Comment:
Any reason that we define two methods to new the payload instance? Can remove this and call `newInstanceFactory()` directly?

Reply:
Just to give meaningful name to wrapper, both are anyways annotated by @ForceInline hence will always get inlined in caller context.

Reply:
The reason that I asked this question is: when I was modifying the VectorMask related code, I will use `VectorPayloadMF.createVectPayloadInstance(...)` when I need to creating a payload for mask. Then I was thinking maybe `VectorPayloadMF.newInstanceFactory()` is better with two reasons:

1. We have already contained "VectorPayloadMF" in the class name.
2. The later is shorter.

Reply:
No problem, please modify as you feel appropriate.